IdentityCore/src/util/ios/MSIDAppExtensionUtil.m

```diff
@@ -68,4 +68,24 @@ + (void)sharedApplicationOpenURL:(NSURL*)url
 #pragma clang diagnostic pop
 }
 
++ (void)sharedApplicationOpenURL:(NSURL *)url
+                         options:(NSDictionary<UIApplicationOpenExternalURLOptionsKey, id> *)options
+               completionHandler:(void (^ __nullable)(BOOL success))completionHandler
+{
+    if ([self isExecutingInAppExtension])
+    {
+        // The caller should do this check but we will double check to fail safely
+        return;
+    }
+    
+    dispatch_async( dispatch_get_main_queue(), ^{
+        
+        SEL openURLSelector = @selector(openURL:options:completionHandler:);
+        UIApplication *application = [self sharedApplication];
+        id (*safeOpenURL)(id, SEL, id, id, id) = (void *)[application methodForSelector:openURLSelector];
+        
+        safeOpenURL(application, openURLSelector, url, options, completionHandler);
```

Comment:
could you explain a little bit on why we cannot call openURL directly? :)

Reply:
Synced offline. Identitycore only allows usage of extension safe APIs. openURL is unsafe API, that's why we're using performSelect in normal openURL. However, performSelector cannot accept 3 parameters.